File: p-e/src/problem1.js
function consecSum(limit) {
  return (limit + 1) * limit / 2
}

function highestMultiple(limit, multiple) {
  return Math.floor(limit / multiple)
}

function sumOfMultiples(limit, multiple) {
  return consecSum(highestMultiple(limit, multiple)) * multiple
}

sumOfMultiples(999,3) + sumOfMultiples(999,5) - sumOfMultiples(999,15)

File: p-e-clj/src/p_e_clj/problem1.clj
(ns p-e-clj.problem1)

(defn consecSum [lim]
  (/ (* (+ lim 1)
        lim)
      2))

(defn highestMultiple [lim, multiple]
  (quot lim multiple))

(defn sumOfMultiples [lim, multiple]
  (* (consecSum (highestMultiple lim multiple))
      multiple))


(-  (+ (sumOfMultiples 999 3)
    (sumOfMultiples 999 5))
    (sumOfMultiples 999 15))